win/tty/wintty.c

```diff
@@ -3513,11 +3518,37 @@ tty_nhgetch(void)
         i = randomkey();
     } else {
 #ifdef UNIX
+#ifdef REALTIME_ON_BOTL
+        fd_set read_set;
```

Comment:
i think something like

```
struct pollfd pf;
pf.fd = 0;
pf.events = POLLIN;
for (;;) {
        if (poll(&pf, 1, 1000)) {
               tgetch stuff
        } else {
                update stats
```

Reply:
oooo u can give the poll thing a timer
got it